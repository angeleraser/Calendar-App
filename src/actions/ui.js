import { TYPES } from "../types/types";

export const openModal = () => ({
  type: TYPES.uiOpenModal,
});

export const closeModal = () => ({
  type: TYPES.uiCloseModal,
});

export const eventClearActiveEvent= () => {
  return {
    type: TYPES.eventClearActiveEvent,
  };
};
