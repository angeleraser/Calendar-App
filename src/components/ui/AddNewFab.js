import React from "react";
import { useDispatch } from "react-redux";
import { openModal } from "../../actions/ui";

export const AddNewFab = () => {
  const dispatch = useDispatch();
  return (
    <button
      onClick={() => {
        dispatch(openModal());
      }}
      className="fab btn btn-primary">
      <i className="fa fa-plus"></i>
    </button>
  );
};
