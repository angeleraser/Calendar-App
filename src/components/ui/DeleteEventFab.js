import { useDispatch } from "react-redux";
import { eventStartDelete } from "../../actions/calendar";
export const DeleteEventFab = () => {
  const dispatch = useDispatch();
  return (
    <button
      onClick={() => {
        console.log('Click')
        dispatch(eventStartDelete());
      }}
      className="fab-danger btn btn-danger">
      Delete <i className="fa fa-trash"></i>
    </button>
  );
};
