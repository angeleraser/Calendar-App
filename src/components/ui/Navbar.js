import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startLogout } from "../../actions/auth";
import { eventLogoutCleanup } from "../../actions/calendar";
export const Navbar = () => {
  const { name: username } = useSelector(({ auth }) => auth);
  const dispatch = useDispatch();
  return (
    <nav className="navbar navbar-dark bg-dark">
      <span className="navbar-brand">
        <i className="fa fa-user mr-2"></i>
        {username}
      </span>
      <button
        onClick={() => {
          dispatch(startLogout());
          dispatch(eventLogoutCleanup());
        }}
        className="btn btn-outline-danger">
        <i className="fa fa-close"></i> Logout
      </button>
    </nav>
  );
};
