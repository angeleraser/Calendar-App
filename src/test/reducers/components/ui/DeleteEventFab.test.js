import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { DeleteEventFab } from "../../../../components/ui/DeleteEventFab";
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();
const wrapper = render(
  <Provider store={store}>
    <DeleteEventFab />
  </Provider>
);
describe("Testing <DeleteEventFav/> component", () => {
  test("should render correctly", () => {
    expect(wrapper.container).toMatchSnapshot();
  });
});
