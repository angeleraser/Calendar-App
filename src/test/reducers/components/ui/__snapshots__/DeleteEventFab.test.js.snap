// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <DeleteEventFav/> component should render correctly 1`] = `
<div>
  <button
    class="fab-danger btn btn-danger"
  >
    Delete 
    <i
      class="fa fa-trash"
    />
  </button>
</div>
`;
